src/topic.ts

```diff
@@ -0,0 +1,97 @@
+import * as grpc from "@grpc/grpc-js";
+import {TigrisClient} from "./proto/server/v1/api_grpc_pb";
+import * as server_v1_api_pb from "./proto/server/v1/api_pb";
+import {
+	SubscribeRequest as ProtoSubscribeRequest,
+	SubscribeResponse as ProtoSubscribeResponse,
+	PublishRequest as ProtoPublishRequest
+} from "./proto/server/v1/api_pb";
+import {Session} from "./session";
+import {
+	PublishOptions,
+	TigrisCollectionType
+} from "./types";
+import {Utility} from "./utility";
+
+export interface SubscribeCallback<T> {
+	onNext(message: T): void;
+
+	onEnd(): void;
+
+	onError(error: Error): void;
+}
+
+export class Topic<T extends TigrisCollectionType> {
+	private readonly _topicName: string;
+	private readonly _db: string;
+	private readonly _grpcClient: TigrisClient;
+
+	constructor(topicName: string, db: string, grpcClient: TigrisClient) {
+		this._topicName = topicName;
+		this._db = db;
+		this._grpcClient = grpcClient;
+	}
+
+	get topicName(): string {
+		return this._topicName;
+	}
+
+	publishMany(tx?: Session, _options?: PublishOptions, ...messages: Array<T>): Promise<Array<T>> {
+		return new Promise<Array<T>>((resolve, reject) => {
+			const messagesArray = new Array<Uint8Array | string>();
+			for (const message of messages) {
+				messagesArray.push(new TextEncoder().encode(Utility.objToJsonString(message)));
```

Comment:
let's create the `TextEncoder` outside the for.

Reply:
I just copied this pattern from elsewhere. We should probably abstract it out so they can both get improvements like this.